elasticsearch/Dockerfile

```diff
@@ -31,7 +31,8 @@ RUN rpm --import https://packages.elastic.co/GPG-KEY-elasticsearch && \
     yum clean all
 # initial boring elasticsearch.yml is just to enable plugin installation.
 # put the real config in place in a deployment with a ConfigMap.
-ADD elasticsearch.yml logging.yml /usr/share/elasticsearch/config/
+ADD elasticsearch.yml /usr/share/elasticsearch/config/
+ADD index_templates /usr/share/elasticsearch/config/index_templates
```

Comment:
This is one of the problems.  I think this should be 
```
ADD index_templates/ /usr/share/elasticsearch/config/index_templates/
```
that is, they should both end in `/`, so that `docker build` (or the openshift image builder) knows that both the source and destination are directories. For example, from the fluentd pod:
```
ADD configs.d/ /etc/fluent/configs.d/
```
The pod log shows this:
```
Create index template '*.json'
...
> PUT /_template/*.json HTTP/1.1
...
< HTTP/1.1 400 Bad Request
```

This means it is trying to use a file literally named `*.json`.  This for loop:
```
	for template_file in /usr/share/elasticsearch/config/index_templates/*.json
```
One of the annoying quirks of bash is that if there are no files which match `/usr/share/elasticsearch/config/index_templates/*.json`, then `template_file` will get set to the literal value `/usr/share/elasticsearch/config/index_templates/*.json`.  I would suggest using `shopt -o failglob` before the for loop.  This will cause the script to fail if there are no matching files.  For more information, see `man bash` - look for shopt and failglob.